ihmc-high-level-behaviors/src/main/java/us/ihmc/PerceptionAndAutonomyProcess.java

```diff
@@ -246,6 +266,9 @@ public PerceptionAndAutonomyProcess(ROS2Helper ros2Helper,
 
       planarRegionsExtractorThread = new RestartableThrottledThread("PlanarRegionsExtractor", 10.0, this::updatePlanarRegions);
       planarRegionsExtractorThread.start();
+
+      heightMapExtractorThread = new RestartableThrottledThread("HeightMapExtractor", 10.0, this::updateHeightMap);
```

Comment:
What is the best way to thread this guy? @TomaszTB @ds58 I think in the continuous hiking process it gets triggered on each realsense frame. Should we do that here and how is the best way to do that?

Reply:
Hmm there are many ways to do it. An `ExecutorService` comes to my mind. You can create an executor, and in the realsense thread submit the call to `updateHeightMap` and pass in the depth image, reference frame, or whatever data it requires. That way the processing for the height map doesn't block the realsense thread, and it runs each time you get an image from the realsense. 

You could also make a separate class which handles the `updateHeightMap` method and whatever logic surrounds it if you want to keep the `PerceptionAndAutonomyProcess` class shorter.